src/firebolt/async_db/util.py

```diff
@@ -0,0 +1,36 @@
+from httpx import Response
+
+from firebolt.common.urls import DATABASES_URL, ENGINES_URL
+
+
+async def is_db_available(connection, database) -> bool:
+    """Verify if the database exists"""
+    resp = await _filter_request(
+        connection, DATABASES_URL, {"filter.name_contains": database}
+    )
+    resp.raise_for_status()
```

Comment:
nit: Can be moved inside _filter_request

Reply:
I agree, it can always be caught if needed.